powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/provider/PowerAuthAuthenticationProvider.java

```diff
@@ -110,6 +110,16 @@ private PowerAuthApiAuthenticationImpl validateSignatureAuthentication(PowerAuth
                     authentication.getData()
             ));
 
+            // Force signature version during migration when commit endpoint is invoked
+            String requestVersion = authentication.getVersion();
```

Comment:
This is definitely not correct - our authentication provider should not depend on some specific resource ID.

Reply:
... what you probably want to do is to add allowed version list to @PowerAuth annotation, with the default to `["2.0", "3.0"]`.

Reply:
... also, request version format should be validated in header validator that is already present.

Reply:
Let's discuss it. We need to force the `signatureVersion` only in case the migration is committed - in this case the v3 signauture needs to be verified instead of v2 right before the version change. I was trying to reuse the standard signature mechanism for this, however we needed to force the version change somehow.

Another option we have is to handle this logic in the `/pa/migration/commit` endpoint itself. I wanted to avoid doing this, because the downside is that signature verification would be handled both by the authentication provider and by the `/pa/migration/commit` endpoint for this specific case. However it may be a cleaner solution...

Reply:
> ... what you probably want to do is to add allowed version list to @PowerAuth annotation, with the default to `["2.0", "3.0"]`.

The point of `signatureVersion` field is not to allow a specific version, however to force a different version than activation version for verifying the first transaction right before commit of migration. We could also the @PowerAuth annotation for such purpose, however the name should be probably different than `allowed version`. Let's discuss in person, this change is tricky.